src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1557,7 +1557,8 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Bind
                         // Cannot escape out of the current expression, as it's a compiler-synthesized location.
                         expression = new BoundDiscardExpression(node, LocalScopeDepth, type: null);
                     }
-                    else if (node.Identifier.ContextualKind() == SyntaxKind.FieldKeyword &&
+                    else if (!IsInsideNameof &&
```

Comment:
>!IsInsideNameof && [](http://example.com/codeflow?start=29&length=18)

I think it would be better to report an error, but not change what is in scope inside nameof.